inst/tutorials/tidy-03-measurement1/tidy-03-measurement1.Rmd

```diff
@@ -0,0 +1,379 @@
+knitr::opts_chunk$set(echo = FALSE)
+tut_reptitle <- "QSS Tidyverse Tutorial 3: Output Report"
+data(STAR, package = "qss")
+star <- STAR
+
+classcounts <- star %>%
+  group_by(classtype) %>%
+  count()
+```
+
+## Handling Missing Data in R
+
+### Small class size data
+
+
+In this chapter, you'll analyze data from the STAR project, which is a four-year randomized trial on the effectiveness of small class sizes on education performance. The `star` data frame as been loaded into your space so that you can play around with it a bit.
+
+Different from the baseR functions, we will teach you how to use tidyverse to perform the same analysis in this tutorial. 
+
+#### Exercises
+- The `tibble` package is a core part of tidyverse. It allows you to convert your traditional R data frames into tibbles, which are data frames too. But tibbles make it easier and faster to work with tidyverse. You can coerce a current data frame to a tibble with `as_tibble`.
+
+```{r as_tibble-star, exercise = TRUE}
+
+```
+
+```{r as_tibble-star-solution}
+as_tibble(star)
+```
+
+```{r as_tibble-star-check}
+grade_code()
+```
+
+- Use the `str` function on the `star` to see what the data looks like. You can always use `head` function to view the first six rows of the data set.
+
+```{r str-star, exercise = TRUE}
+
+```
+
+```{r str-star-solution}
+str(star)
+```
+
+```{r str-star-check}
+grade_code()
+```
+
+
+- Use the `glimpse` function on the `star` to see what the data looks like. `glimpse` is particularly useful when the data set contains a long list of variables (columns), as it allows you to see every column by transposing the original data set. 
+
+```{r glimpse-star, exercise = TRUE}
+
+```
+
+```{r glimpse-star-solution}
+glimpse(star)
+```
+
+```{r glimpse-star-check}
+grade_code()
+```
+
+- Use the `dim_desc` function on the `star` to see what the dimensions of the data look like.
+
+```{r dim_desc-star, exercise = TRUE}
+
+```
+
+```{r dim_desc-star-solution}
+dim_desc(star)
+```
+
+```{r dim_desc-star-check}
+grade_code()
+```
+
+
+- Use the `summary` function on the `star` to get a sense for each variable.
+
+```{r summary-star, exercise = TRUE}
+
+```
+
+```{r summary-star-solution}
+summary(star)
+```
+
+```{r summary-star-check}
+grade_code()
+```
+
+
+
+### Data wrangling with tidyverse: Handling missing data
+
+From the baseR version of the qss tutorials, you have already learnt the basic toolkit that allows you to identify and work with the missing values (`NA`s) in your data set. 
+
+Tidyverse also provides helpful tools for you to deal with the missing values, and we teach you how to use tidyverse to address the explicit missing values in this tutorial.  
+
+#### Exercises
+- Use the `drop_na` functions to drop the rows with explicit missing values in variable `g4math`. However, it is vital to make sure that you are not dropping the observations that are important in other representations of the data. So be careful when you use `drop_na`.
```

Comment:
Yes! drop_na() is useful but causes listwise deletion so I wanted to remind the students to be aware of the possibility of accidentally dropping observations in other variables, and I guess I should not have used this term representation here since it is quite ambiguous indeed. I have tried to explain more specifically so please double check on this one! Thank you!